client/cody-shared/src/sourcegraph-api/graphql/client.ts

```diff
@@ -84,6 +89,10 @@ export class SourcegraphGraphQLAPIClient {
         this.config = newConfig
     }
 
+    public isDotCom(): boolean {
+        return this.config.serverEndpoint === this.dotcomUrl
```

Comment:
Instead of doing string equality, can we compare the origin linke this so eventual trailing `/` will not break the logic? 

```suggestion
        return new URL(this.config.serverEndpoint).origin === new URL(this.dotcomUrl).origin
```